### PR TITLE
One event class and Event Broker config extension

### PR DESCRIPTION
**Pull request by pdyba**

### CODE REVIEW

File: lbz/events/api.py [hunk 1]
```diff
@@ -60,28 +43,28 @@ def set_bus_name(self, bus_name: str) -> None:
...
+    def failed_events(self) -> List[Event]:
         return deepcopy(self._failed_events)
 
-    def register(self, new_event: BaseEvent) -> None:
+    def register(self, new_event: Event) -> None:
         self._pending_events.append(new_event)
 
     # TODO: Stop sharing protected lists outside the class, use the above properties instead
-    def get_all_pending_events(self) -> List[BaseEvent]:
+    def get_all_pending_events(self) -> List[Event]:
```
Comment (redlickigrzegorz) [inline lbz/events/api.py:61]: After your changes, we will need to adjust the rest of our code anyway, so I think this is a good moment to remove these methods 😉

File: lbz/events/broker.py [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
-    data: dict
-
-
 class EventBroker:
-    def __init__(self, mapper: Dict[str, List[Callable]], raw_event: dict) -> None:
+    def __init__(
+        self,
+        mapper: Dict[str, List[Callable]],
+        raw_event: dict,
+        type_key: str = "detail-type",
```
Comment (redlickigrzegorz) [inline lbz/events/broker.py:14]: Make these new arguments keyword-only - it is very easy to assign them wrong using positional:
```python
def __init__(
    self,
    mapper: Dict[str, List[Callable]],
    raw_event: dict,
    *,
    type_key: str = "detail-type",
    data_key: str = "detail",
) -> None:
```

File: setup.cfg [hunk 3]
```diff
@@ -24,8 +24,8 @@ use_parentheses = True
 [mypy]
 # https://mypy.readthedocs.io/en/latest/running_mypy.html#mapping-file-paths-to-modules
 explicit_package_bases = True
-mypy_path =
-  $MYPY_CONFIG_FILE_DIR,
+;mypy_path =
```
[the remark this answers is not in the document]
Comment (pdyba) [reply setup.cfg:27]: I could not run mypy in newest version with that setting it was throwing:

> 
> mypy 0.971 (compiled: yes)
> /Users/dybapi/PycharmProjects/lambdalizator is in the MYPYPATH. Please remove it.
> See https://mypy.readthedocs.io/en/stable/running_mypy.html#how-mypy-handles-imports for more info
> make: *** [mypy] Error 1
> 

Can You reproduce it or is it just my system issue ?

File: lbz/events/event.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        self.data = data
+        self.type = event_type or self.type
+
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, Event):
+            return self.type == other.type and self.data == other.data
+        return False
+
+    def __repr__(self) -> str:
+        return f"Event(type='{self.type}', data={self.data})"
```
[the remark this answers is not in the document]
Comment (pdyba) [reply lbz/events/event.py:18]: It was there out of the box in the original data class.

File: lbz/events/api.py (hunk 1 above)
Comment (pdyba) [reply lbz/events/api.py:61]: I think we agreed to do it with 0.6.0

Comment (pdyba) [reply lbz/events/api.py:61]: And as far as I checked it should not be a breaking change.

File: setup.cfg (hunk 3 above)
[the remark this answers is not in the document]
Comment (redlickigrzegorz) [reply setup.cfg]: According to the discussion, this error may be related to the changes in [mypy v0.971](https://mypy-lang.blogspot.com/2022/07/mypy-0971-released.html):
> Fix editable installs to current working directory (Shantanu, PR [13161](https://github.com/python/mypy/pull/13161))

We decided to remove this option from the configuration.

File: lbz/events/event.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (redlickigrzegorz) [reply lbz/events/event.py:18]: But that does not mean it was good, dataclass puts there everything by default, I think.

If we need to present data, we still can do this but I would keep representation clearer 😉

**Review by redlickigrzegorz — APPROVED** [review]

**Pull request merged**